autogpt_platform/backend/backend/integrations/oauth/device_base.py

```
@@ -0,0 +1,115 @@
+
+    def needs_refresh(self, credentials: OAuth2Credentials) -> bool:
+        """Indicates whether the given tokens need to be refreshed."""
+        return (
+            credentials.access_token_expires_at is not None
+            and credentials.access_token_expires_at < int(time.time()) + 300
+        )
+
+    def handle_default_scopes(self, scopes: list[str]) -> list[str]:
+        """Uses default scopes when none are provided."""
+        if not scopes:
+            logger.debug(f"Using default scopes for provider {str(self.PROVIDER_NAME)}")
```

Ruff (0.16.1)

[warning] Use explicit conversion flag

Replace with conversion flag

(RUF010)